### ISSUE TITLE
Configure health monitoring alerts

### ISSUE THREAD

**Issue by adunkman** [ustaxcourt/ef-cms#250]
As the Court, so that we can ensure we have a secure and available system, we need assurance that after code updates, known weak points are not vulnerable to leak sensitive data and the application is available.

Acceptance criteria

Alerts are configured when: 

- [x] The UI is unavailable. Monitored by alerting on uptime/ping testing for dawson.ustaxcourt.gov and app.dawson.ustaxcourt.gov (and the equivalent in other environments). Implemented in #497.
- [x] The system health endpoints return red. Monitored by alerting on the system health endpoints (as implemented in flexion#6281). Blocked on https://github.com/flexion/ef-cms/issues/6903. Implemented in https://github.com/ustaxcourt/ef-cms/pull/650.
- [x] The Elasticsearch cluster has a non-green status. Monitored by alerting on the cluster health status. Implemented in #539.

**Comment by adunkman** [comment]
Blocked by ustaxcourt/ef-cms-flexion#387.

**Comment by adunkman** [comment]
Had a quick chat with @julialeague about this to help set my head straight on a path forward — thanks Julia!!

There are a lot of things we could be looking at, but without knowing usage patterns well (#137), we can’t know what unusual looks like. Therefore, I propose: 

- This issue tackles the obvious fires. These are things we know would mean an outage or a problem. They include: 
  - The UI is unavailable. Monitored by alerting on uptime/ping testing for dawson.ustaxcourt.gov and app.dawson.ustaxcourt.gov (and the equivalent in other environments).
  - The system health endpoints return red. Monitored by alerting on the system health endpoints (as implemented in https://github.com/flexion/ef-cms/issues/6281).
  - The Elasticsearch cluster has a non-green status. Monitored by alerting on the cluster health status. 

At a future point, once we know what "normal" traffic looks like, we can consider monitoring things like: 
- Unexpected traffic volume.
- Login or new signup rates abnormal.

**Comment by adunkman** [comment]
> - Configured S3 buckets are not publicly available.
> - Elasticsearch cannot be accessed directly.

Speaking to these points, given the rate of change of these, I think they’d result in fragile tests. Considering that the "pass" state is a URL is inaccessible, I think we’d quickly be asserting things which were no longer of help (for example, that a nonsense URL was inaccessible).

Instead, I think we might want to consider introducing a security scanner for these. GitHub’s [super-linter](https://github.com/github/super-linter) seems to be a good option, which uses [tflint](https://github.com/terraform-linters/tflint) for identifying formatting/known linter problems and [terrascan](https://github.com/accurics/terrascan) for identifying security risks.

I’ll file a new issue for these, for post-MVP.

**Comment by adunkman** [comment]
> The UI is unavailable. Monitored by alerting on uptime/ping testing for dawson.ustaxcourt.gov and app.dawson.ustaxcourt.gov (and the equivalent in other environments).

I believe this can be accomplished with [Route53 health checks](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/monitoring-health-checks.html) which would trigger a CloudWatch alarm.

> The system health endpoints return red. Monitored by alerting on the system health endpoints (as implemented in flexion#6281).

This can be achieved by a Route53 health check as well, hitting the health check endpoint on the public API. Unfortunately, we no longer have a set URL for the public API — it will be either https://public-api-green.dawson.ustaxcourt.gov/public-api/health or https://public-api-blue.dawson.ustaxcourt.gov/public-api/health. I’ll need to file an issue (closely related to https://github.com/flexion/ef-cms/issues/6864) and have it fixed before I can fully implement this health check.

> The Elasticsearch cluster has a non-green status. Monitored by alerting on the cluster health status.

Already handled by a CloudWatch metric, and we can add an alarm.

**Comment by adunkman** [comment]
CloudWatch Alarms looks like a natural clearing house for status, and it uses SNS for notifications.

**Comment by adunkman** [comment]
Speaking with @mmarcotte on direction here — we’ll use a simple SNS configuration for now and go with the Route53 approach. 

We know we have a blindspot that if AWS has a catastrophic outage, we will not be notified (because the notification system may also be offline), and I’ll file an issue to consider using an external notification service like [Opsgenie](https://www.atlassian.com/software/opsgenie) for post-MVP.

**Comment by adunkman** [comment]
Reported https://github.com/flexion/ef-cms/issues/6903 to get a single API endpoint for the system health JSON.

**Comment by adunkman** [comment]
~Speaking w/ Mike, he’d like the API endpoint covered by health alerts for the first pass as well. Updating the description!~ [I misunderstood](https://ustaxcourt.slack.com/archives/D016Z82DQ7R/p1604084108017000?thread_ts=1604081539.014800&cid=D016Z82DQ7R).

**Comment by adunkman** [comment]
Latest Elasticsearch alarms are in https://github.com/ustaxcourt/ef-cms/compare/add-es-alarms, blocked on running account-specific terraform steps, [communication in Slack](https://ustaxcourt.slack.com/archives/C015SEE9CUU/p1604589853205400).

**Comment by adunkman** [comment]
https://github.com/flexion/ef-cms/issues/6903 is completed by https://github.com/flexion/ef-cms/pull/7177, awaiting PR to the Court.

**Comment by adunkman** [comment]
Awaiting https://github.com/ustaxcourt/ef-cms/pull/608.
